### PR TITLE
support for feature reports on HID interace

### LINKED ISSUE #363: HID feature reports
Placeholder to add HID feature reports.

Source: https://docs.microsoft.com/en-us/windows-hardware/drivers/hid/introduction-to-hid-concepts

| Name | Description |
|------|------|
| **Feature Report** | Data blobs that can be manually read and/or written, and are typically related to configuration information. |

### PR DESCRIPTION
**Pull request by koltenpearson**
should fix #363 and add support for HID feature reports

this has been tested briefly on linux, still needs to be tested on windows and mac

this only adds functionality to the HID interface, it does not at it to the USB one

### CODE REVIEW

**Review by bberenz — CHANGES_REQUESTED** [review]

File: src/qz/communication/PJHA_HidIO.java [hunk 1]
```diff
@@ -110,6 +110,32 @@ public void sendData(byte[] data, Byte reportId) throws DeviceException {
...
+
+    public void sendFeatureReport(byte[] data, Byte reportId) throws DeviceException {
+        if (reportId == null) { reportId = (byte)0x00; }
+
+        byte[] report = new byte[1 + data.length];
+        report[0] = reportId;
+        System.arraycopy(data, 0, report, 1, data.length);
+
+        //NOTE this form of this method is deprecated in most recent version of purejavahidapi
+        //once the dependency is updated, the new version accepts the report id separately from the data
```
[the remark this answers is not in the document]
Comment (koltenpearson) [reply src/qz/communication/PJHA_HidIO.java:130]: looks like you updated it to version 0.0.3, but their github repo shows up to 0.0.10, with the deprecation currently on master and not yet tagged
https://github.com/nyholku/purejavahidapi

so I guess not

[the remark this answers is not in the document]
Comment (bberenz) [reply src/qz/communication/PJHA_HidIO.java:130]: oh, you know what, it looks like they never updated their [pom.xml](https://github.com/nyholku/purejavahidapi/blob/master/pom.xml#L7) versioning
we should have the latest on their master though, as it was built from source

[the remark this answers is not in the document]
Comment (tresf) [reply src/qz/communication/PJHA_HidIO.java:130]: Good catch @bberenz.  Can you patch this on our `master` branch so we aren't confused down the road?

[the remark this answers is not in the document]
Comment (tresf) [reply src/qz/communication/PJHA_HidIO.java:130]: FYI, also requested the change upstream: https://github.com/nyholku/purejavahidapi/pull/77

[the remark this answers is not in the document]
Comment (koltenpearson) [reply src/qz/communication/PJHA_HidIO.java]: made the adjustments to the method, and it compiles successfully

**Review by bberenz — APPROVED** [review]

**Pull request merged**